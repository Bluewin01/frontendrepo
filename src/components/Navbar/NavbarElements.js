import styled from "styled-components";
import { NavLink as Link } from "react-router-dom";

export const Nav = styled.nav`
  background: #d31145;
  height: 65px;
  display: flex;
  justify-content: space-between;
  padding: 0 calc((80vw - 1000px) / 2);
  z-index: 10;
`;

export const NavLogo = styled(Link)`
  color: #fff;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  font-size: 1.8rem;
  font-weight: bold;

  &.active {
    color: #fff;
  }

  img {
    width: 44px;
  }
`;

export const NavMenu = styled.nav`
  display: flex;
  align-items: center;
  width: 100vw;
  white-space: nowrap;

  @media screen and (max-width: 768px) {
    display: none;
  }
`;

export const NavLink = styled(Link)`
  color: #fff;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  font-size: 2.5rem;
  font-weight: bold;

  &.active {
    color: #fff;
  }

  &:hover {
    background-color: rgba(85, 67, 68, 0.4);
    text-decoration: none;
    color: #fff;
  }

  img {
    width: 44px;
  }
`;

export const NavBtn = styled.nav`
  display: flex;
  align-items: center;
  margin-right: 30px;

  @media screen and (max-width: 768px) {
    display: none;
  }
`;

export const NavBtnLink = styled(Link)`
  border-radius: 4px;
  background: #d31145;
  padding: 10px 40px;
  color: #fff;
  border: none;
  outline: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  text-decoration: none;
  font-size: 1.2rem;
  font-weight: bold;
  box-shadow: 0 3px 10px rgb(0 0 0 / 0.5);

  &:hover {
    text-decoration: none;
    transition: all 0.2s ease-in-out;
    background: #fff;
    color: #d31145;
  }
`;

export const Logout = styled.button`
  display: flex
  align-items: center;
  background-color: #d31145;
  box-shadow: 0 3px 8px 0 rgb(4 0 0 / 50%);
  border: none;
  border-radius: 10px;
  color: white;
  padding: 0px 10px 0px 10px;
  font-weight: 500;
  height: 40px;

  &:hover {
    transition: all 0.2s ease-in-out;
    color: #dce2e994;
  }
`;
