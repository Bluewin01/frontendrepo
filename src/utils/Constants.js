export const theme = {
  color: {
    w: "#FFFFFF",
    k: "#000000",
    p1: "#D31145",
    p2: "#596C80",
    p3: "#554344",
    p4: "#363E3F",
    p5: "#e6edeb",
    s1: "#E86487",
    s2: "#FBBF56",
    s3: "#97CB5D",
    s4: "#46D3BD",
    s5: "#22A8DA",
    s6: "#9962D2",
    t1: "#F6CCD7",
    t2: "#B8123E",
    t3: "#F9E1B8",
    t4: "#C36418",
    t5: "#C6EB9E",
    t6: "#65962E",
    b1: "#D4EDF1",
    b2: "#FAF9F3",
    b3: "#F0EEE1",
    b4: "#DBDAD4",
    b5: "#B3B1A8",
    b6: "#6F6E68",
    b8: "#302F2C",
    b9: "#292025",
    u2: "#FF5500",
    u3: "#405B70",
    u4: "#769FAB",
  },
};

export const BASE_API = `${process.env.REACT_APP_BASE_API_URL}`;
