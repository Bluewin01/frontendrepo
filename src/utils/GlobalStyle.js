import { createGlobalStyle } from 'styled-components';
import 'antd/dist/antd.css';
import '@aia-digital/ui-library/lib/static/styles.css';

import AIABoldEOT from '../assets/fonts/AIAEverest-Bold.eot';
import AIABoldWOFF from '../assets/fonts/AIAEverest-Bold.woff';
import AIABoldWOFF2 from '../assets/fonts/AIAEverest-Bold.woff2';
import AIABoldTTF from '../assets/fonts/AIAEverest-Bold.ttf';
import AIABoldSVG from '../assets/fonts/AIAEverest-Bold.svg';

import AIACondensedEOT from '../assets/fonts/AIAEverest-Condensed.eot';
import AIACondensedWOFF from '../assets/fonts/AIAEverest-Condensed.woff';
import AIACondensedWOFF2 from '../assets/fonts/AIAEverest-Condensed.woff2';
import AIACondensedTTF from '../assets/fonts/AIAEverest-Condensed.ttf';
import AIACondensedSVG from '../assets/fonts/AIAEverest-Condensed.svg';

import AIACondensedMediumEOT from '../assets/fonts/AIAEverest-CondensedMedium.eot';
import AIACondensedMediumWOFF from '../assets/fonts/AIAEverest-CondensedMedium.woff';
import AIACondensedMediumWOFF2 from '../assets/fonts/AIAEverest-CondensedMedium.woff2';
import AIACondensedMediumTTF from '../assets/fonts/AIAEverest-CondensedMedium.ttf';
import AIACondensedMediumSVG from '../assets/fonts/AIAEverest-CondensedMedium.svg';

import AIAExtraBoldEOT from '../assets/fonts/AIAEverest-ExtraBold.eot';
import AIAExtraBoldWOFF from '../assets/fonts/AIAEverest-ExtraBold.woff';
import AIAExtraBoldWOFF2 from '../assets/fonts/AIAEverest-ExtraBold.woff2';
import AIAExtraBoldTTF from '../assets/fonts/AIAEverest-ExtraBold.ttf';
import AIAExtraBoldSVG from '../assets/fonts/AIAEverest-ExtraBold.svg';

import AIAMediumEOT from '../assets/fonts/AIAEverest-Medium.eot';
import AIAMediumWOFF from '../assets/fonts/AIAEverest-Medium.woff';
import AIAMediumWOFF2 from '../assets/fonts/AIAEverest-Medium.woff2';
import AIAMediumTTF from '../assets/fonts/AIAEverest-Medium.ttf';
import AIAMediumSVG from '../assets/fonts/AIAEverest-Medium.svg';

import AIARegularEOT from '../assets/fonts/AIAEverest-Regular.eot';
import AIARegularWOFF from '../assets/fonts/AIAEverest-Regular.woff';
import AIARegularWOFF2 from '../assets/fonts/AIAEverest-Regular.woff2';
import AIARegularTTF from '../assets/fonts/AIAEverest-Regular.ttf';
import AIARegularSVG from '../assets/fonts/AIAEverest-Regular.svg';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'AIABold';
    src: url('${AIABoldEOT}');
    src: url('${AIABoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIABoldWOFF}') format('woff'),
        url('${AIABoldWOFF2}') format('woff2'),
        url('${AIABoldTTF}') format('truetype'),
        url('${AIABoldSVG}#AIAEverestBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensed';
    src: url('${AIACondensedEOT}');
    src: url('${AIACondensedEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedWOFF}') format('woff'),
        url('${AIACondensedWOFF2}') format('woff2'),
        url('${AIACondensedTTF}') format('truetype'),
        url('${AIACondensedSVG}#AIAEverestCondensed') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensedMedium';
    src: url('${AIACondensedMediumEOT}');
    src: url('${AIACondensedMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedMediumWOFF}') format('woff'),
        url('${AIACondensedMediumWOFF2}') format('woff2'),
        url('${AIACondensedMediumTTF}') format('truetype'),
        url('${AIACondensedMediumSVG}#AIAEverestCondensedMedium') format('svg');
    font-weight: 500;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAExtraBold';
    src: url('${AIAExtraBoldEOT}');
    src: url('${AIAExtraBoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIAExtraBoldWOFF}') format('woff'),
        url('${AIAExtraBoldWOFF2}') format('woff2'),
        url('${AIAExtraBoldTTF}') format('truetype'),
        url('${AIAExtraBoldSVG}#AIAEverestExtraBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAMedium';
    src: url('${AIAMediumEOT}');
    src: url('${AIAMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIAMediumWOFF}') format('woff'),
        url('${AIAMediumWOFF2}') format('woff2'),
        url('${AIAMediumTTF}') format('truetype'),
        url('${AIAMediumSVG}#AIAEverestMedium') format('svg');
    font-weight: 600;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIARegular';
    src: url('${AIARegularEOT}');
    src: url('${AIARegularEOT}?#iefix') format('embedded-opentype'),
        url('${AIARegularWOFF}') format('woff'),
        url('${AIARegularWOFF2}') format('woff2'),
        url('${AIARegularTTF}') format('truetype'),
        url('${AIARegularSVG}#AIAEverestRegular') format('svg');
    font-weight: 400;
    font-style: normal;
  }

  body h1 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 96px;
    line-height: 84px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h2 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 72px;
    line-height: 64px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h3 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 48px;
    line-height: 44px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h4 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 32px;
    line-height: 28px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h5 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 24px;
    line-height: 22px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h6 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body p, ul, ol {
    font-family: "AIARegular", Arial, "Helvetica Neue", Helvetica, sans-serif;
    font-size: 15px;
    line-height: 20px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }
`;

export default GlobalStyle;
