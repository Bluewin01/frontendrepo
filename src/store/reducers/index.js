import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
