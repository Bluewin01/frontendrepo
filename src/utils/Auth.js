// Service to check authentication for user and to signOut
import * as AuthActions from "../store/Auth/AuthActions";
import store from "../store";

const Auth = {
  signOut() {
    store.dispatch(AuthActions.signOut);
  },
  isAuth(jwt) {
    if (!jwt) {
      return false;
    } else {
      return true;
    }
  },
};
export default Auth;
