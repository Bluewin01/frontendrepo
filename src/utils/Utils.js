// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
