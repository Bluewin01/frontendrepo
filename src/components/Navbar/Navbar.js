import React from "react";
import {
  Nav,
  NavLogo,
  NavMenu,
  NavLink,
  NavBtn,
  Logout,
} from "./NavbarElements";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { makeSelectJwt } from "../../store/Auth/AuthReselect";
import { createStructuredSelector } from "reselect";
import { signOut } from "../../store/Auth/AuthActions";
import AIALogo from "./../../assets/images/aia-logo-white.png";
import store from "../../store";
import { MdOutlineLogout } from "react-icons/md";

const Navbar = (props) => {
  const { jwt } = props;

  const handleLogout = () => {
    store.dispatch(signOut());
  };

  return (
    <Nav>
      <NavLogo to="/" data-testid="navlogo">
        <img src={AIALogo} alt="aia-logo-white" />
      </NavLogo>
      <NavMenu>
        <NavLink to="/">Notification Template Portal</NavLink>
      </NavMenu>
      {jwt === null ? null : (
        <NavBtn>
          <Logout type="primary" size="large" danger onClick={handleLogout}>
            <MdOutlineLogout size={24} /> Logout
          </Logout>
        </NavBtn>
      )}
    </Nav>
  );
};

const mapStateToProps = createStructuredSelector({
  jwt: makeSelectJwt(),
});

Navbar.propTypes = {
  jwt: PropTypes.object,
};

export default connect(mapStateToProps)(Navbar);
