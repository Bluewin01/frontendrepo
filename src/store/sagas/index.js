import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
