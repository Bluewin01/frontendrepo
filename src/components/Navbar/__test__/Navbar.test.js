import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, screen } from "../../../utils/Utils";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../../store";
import { Provider } from "react-redux";
import Navbar from "../Navbar";

describe("Component: Navbar", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it("should be render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
