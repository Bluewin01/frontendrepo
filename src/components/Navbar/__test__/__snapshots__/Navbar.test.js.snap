// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Navbar should be render correctly 1`] = `
<nav
  className="sc-bdVaJa beqmPi"
>
  <a
    aria-current="page"
    className="sc-bwzfXH ieUGJf active"
    data-testid="navlogo"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="aia-logo-white"
      src={Object {}}
    />
  </a>
  <nav
    className="sc-htpNat WimjQ"
  >
    <a
      aria-current="page"
      className="sc-bxivhb nGEPR active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Notification Template Portal
    </a>
  </nav>
</nav>
`;
